### PR TITLE
[Tizen] Run matter unit tests on QEMU

### PR DESCRIPTION
**Pull request by arkq**
### Changes

- add `chip_pw_run_tests` define to allow disabling pigweed test runner but still be able to link test executables (if someone has a better idea how to achieve such a goal, I'm all ears :))
- run Matter unit tests on Tizen platform in QEMU
- remove duplicated example builds from examples-tizen workflow (these examples are built by QEMU workflow)
- document how to debug tests on Tizen QEMU
- added "Run test (Linux x64)" to vscode debug menu for easier debugging

### Testing

CI will test.

### CODE REVIEW

**Comment by woody-apple** [comment]
@andy31415 To review the common parts.

File: .github/.wordlist.txt [hunk 1]
```diff
@@ -1157,6 +1157,7 @@ rmw
 rodata
 Rollershade
 rootfs
+rootshell
```
Comment (andy31415) [inline .github/.wordlist.txt:1160]: switch quoting to backtics so this is not needed as a word

File: build/chip/tests.gni [hunk 2]
```diff
@@ -20,16 +20,21 @@ import("${chip_root}/src/platform/device.gni")
 declare_args() {
   # Enable building tests.
   chip_build_tests = current_os != "freertos"
+
+  # Use source_set instead of static_lib for tests.
```
[the remark this answers is not in the document]
Comment (arkq) [reply build/chip/tests.gni]: This define was already there (it was added by https://github.com/project-chip/connectedhomeip/pull/8842 and I guess it is required for EFR32 tests to run properly). But when adding `chip_pw_run_tests` I've reordered other defines... I've reverted that reorder, so the patch will be cleaner now.

File: .github/.wordlist.txt (hunk 1 above)
Comment (arkq) [reply .github/.wordlist.txt]: Done

**Review by andy31415 — APPROVED** [review]

**Pull request merged**